packages/xstate-test/src/machine.ts

```diff
@@ -0,0 +1,135 @@
+import { serializeState, SimpleBehavior } from '@xstate/graph';
+import {
+  ActionObject,
+  AnyEventObject,
+  AnyState,
+  AnyStateMachine,
+  createMachine,
+  EventFrom,
+  EventObject,
+  StateFrom,
+  TypegenConstraint,
+  TypegenDisabled
+} from 'xstate';
+import { TestModel } from './TestModel';
+import {
+  TestMachineConfig,
+  TestMachineOptions,
+  TestModelEventConfig,
+  TestModelOptions
+} from './types';
+import { flatten } from './utils';
+import { validateMachine } from './validateMachine';
+
+export async function testStateFromMeta(state: AnyState) {
+  for (const id of Object.keys(state.meta)) {
+    const stateNodeMeta = state.meta[id];
+    if (typeof stateNodeMeta.test === 'function' && !stateNodeMeta.skip) {
+      await stateNodeMeta.test(state);
+    }
+  }
+}
+
+export function createTestMachine<
+  TContext,
+  TEvent extends EventObject = AnyEventObject,
+  TTypesMeta extends TypegenConstraint = TypegenDisabled
```

Comment:
are we going to have a separate type of typegen for test machines?

Reply:
It can work off the same definition, since this is a subset of the original createMachine.

We should add `createTestMachine` to the list of functions that `@xstate/machine-extractor` attempts to parse.